magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL2_to_DL3.py

```diff
@@ -178,7 +184,20 @@ def main():
         help="Path to a configuration file",
     )
 
+    parser.add_argument(
+        "--dense_MC_sources",
+        "-d",
+        dest="dense_list",
+        type=str,
+        help="File with name of sources to be processed with the dense MC train line",
+    )
+
     args = parser.parse_args()
+    dense_list = []
+    if args.dense_list is not None:
+        with open(args.dense_list) as d:
+            dense_list = d.readlines()
```

Comment:
if you do like this, the list will have '\n' at the end of each position, and then the source names will not be matched (unless they include '\n' as well) - have you checked if this works?
you can get read of '\n' easily if you do this instead:
```
dense_list=d.read().splitlines()
```